File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.cs
using System;

namespace CodingTest
{
    internal class Program
    {
        static string[] s = Console.ReadLine().Split();
        static int n = int.Parse(s[0]);
        static int m = int.Parse(s[1]);
        static int v = int.Parse(s[2]);
        static int[] visitedDFS = new int[n + 1];
        static int[] visitedBFS = new int[n + 1];
        static List<int>[] nearNode = new List<int>[n + 1];

        static void DFS(int node)
        {
            if (visitedDFS[node] == 1) return;
            visitedDFS[node] = 1;
            Console.Write(node + " ");

            for (int i = 0; i < nearNode[node].Count; i++)
            {
                int x = nearNode[node][i];
                DFS(x);
            }
        }
        static void Main(string[] args)
        {
            Queue<int> bfs = new Queue<int>();
            for (int i = 0; i < nearNode.Length; i++)
            {
                nearNode[i] = new List<int>();
            }

            for (int i = 0; i < m; i++)
            {
                string[] ss = Console.ReadLine().Split();
                int a = int.Parse(ss[0]);
                int b = int.Parse(ss[1]);
                nearNode[a].Add(b);
                nearNode[b].Add(a);
            }

            for (int i = 0; i < nearNode.Length; i++)
            {
                nearNode[i].Sort();
            }

            DFS(v);
            Console.WriteLine();
            bfs.Enqueue(v);
            visitedBFS[v] = 1;
            while (bfs.Count != 0)
            {
                int node = bfs.Dequeue();
                Console.Write(node + " ");

                for (int i = 0; i < nearNode[node].Count; i++)
                {
                    int x = nearNode[node][i];
                    if (visitedBFS[x] != 1)
                    {
                        bfs.Enqueue(x);
                        visitedBFS[x] = 1;
                    }
                }
            }
        }
    }
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.cc
#include <iostream>
#include <vector>
#include <queue>
#include <algorithm>
using namespace std;

int n, m, v;
vector<vector<int>> graph;
vector<bool> visited;

void dfs(int x) {
    visited[x] = true;
    cout << x << " ";

    for (int i = 0; i < graph[x].size(); i++)
    {
        int y = graph[x][i];
        if (!visited[y]) dfs(y);
    }
}

void bfs() {
    vector<bool> visited(n + 1);
    queue<int> q;
    q.push(v);
    visited[v] = true;

    while (!q.empty())
    {
        int currentNode = q.front();
        q.pop();
        cout << currentNode << " ";

        for (int node : graph[currentNode])
        {
            if (!visited[node])
            {
                q.push(node);
                visited[node] = true;
            }
        }
    }
}

int main(void) {
    cin >> n >> m >> v;
    graph.resize(n + 1);
    visited.resize(n + 1);
    for (int i = 0; i < m; i++)
    {
        int a, b;
        cin >> a >> b;
        graph[a].push_back(b);
        graph[b].push_back(a);
    }

    for (int i = 1; i <= n; i++) 
    {
        sort(graph[i].begin(), graph[i].end());
    }

    dfs(v);
    cout << "\n";
    bfs();

    return 0;
}